### PR TITLE
Update flexget to 2.9.14. Request #2551

### PR DESCRIPTION
**Pull request by ymartin59**
- No longer use timestamp in package version
- Enforce flexget version at packaging
- Add lacking functools32 dependency

_Linked issues:_ #2551 

### Checklist
- [X] Build rule `all-supported` completed successfully
- [ ] ~~Package upgrade completed successfully~~. Decided to change project versioning, so uninstallation is required to upgrade
- [X] New installation of package completed successfully. Used XPEnologoy DS3615xs `5.2-5967.1`

**Warning**: I retrieved upstream `requirements.txt` file and include `flexget` and  `functools32` to get service starting but I fear other transitive dependencies are lacking for specific features to work. A set of test cases to apply would be welcome. I left *beta" flag for this reason, propose to remove it after "real user" feedback.

`xpenology52> /usr/local/flexget/env/bin/flexget --version`
`2.9.14`
`You are on the latest release.`

### CODE REVIEW

File: spk/flexget/src/requirements.txt [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
-FeedParser>=5.1.3
-argparse
-babelfish>=0.5.4
-stevedore>=0.14
-BeautifulSoup
-beautifulsoup4>=4.1, !=4.2.0, <4.4
-plumbum
-rpyc
-path.py
+FeedParser>=5.2.1
```
Comment (Dr-Bean) [inline spk/flexget/src/requirements.txt]: See the WIKI article on wheels: https://github.com/SynoCommunity/spksrc/wiki/Using-wheels-to-distribute-Python-packages. 
All wheels must be pinned to a specific version, and all requirements should be listed here (unless they are providedvia the Python package, e.g. setuptools, wheel, pip, other included cross-compiled modules, and modules listed in Pythones requirements.txt).

For Flexget, the current complete list looks like this:
```
aniso8601==1.2.0
APScheduler==3.3.1
babelfish==0.5.5
beautifulsoup4==4.5.3
cheroot==5.1.0
CherryPy==10.0.0
click==6.7
colorclass==2.2.0
feedparser==5.2.1
Flask==0.12
Flask-Compress==1.4.0
Flask-Cors==3.0.2
Flask-Login==0.4.0
Flask-RESTful==0.3.5
flask-restplus==0.8.6
FlexGet==2.9.11
funcsigs==1.0.2
functools32==3.2.3.post2
future==0.16.0
futures==3.0.5
guessit==2.0.4
html5lib==0.999999999
itsdangerous==0.24
Jinja2==2.9.5
jsonschema==2.5.1
MarkupSafe==0.23
path.py==10.1
pathlib==1.0.1
plumbum==1.6.3
portend==1.8
pynzb==0.1.0
pyparsing==2.1.10
PyRSS2Gen==1.1
python-dateutil==2.6.0
pytz==2016.10
PyYAML==3.12
rebulk==0.8.2
requests==2.13.0
rpyc==3.3.0
six==1.10.0
SQLAlchemy==1.1.5
tempora==1.6.1
terminaltables==3.1.0
tzlocal==1.3
webencodings==0.5
Werkzeug==0.11.15
wheel==0.24.0
zxcvbn-python==4.4.14
```
As mentioned above, note that there may be modules that need cross-compiling in this list, as well as default modules provided by the Python package. Those may need to be excluded/commented out (e.g. PyYAML is one of those).

Comment (ymartin59) [inline spk/flexget/src/requirements.txt]: So my initial fear about dependency management was right... Please let me a couple of days to learn how to get it right.

**Comment by ymartin59** [comment]
@Dr-Bean Is there any guideline or support to run tests so to validate tool deployed by package installation is fully functional ? For Flexget specific case, I plan to run "py.test" to confirm dependencies deployments. I may do it manually but it is worth to be documented or automated for future. Any advice ?

**Comment by Dr-Bean** [comment]
Good question. Not sure I can give a satisfactory answer though :(
Unfortunately, there's no standard way to check if a package works as intended. It depends on the type of software (e.g. compiled, a simple copy/paste or anything in between), and what the software itself does and provides.
There are two considerations here: the package, and the software itself.
For the first, documentation is available in the form of Synology package documentation and what we have in the wiki (as limited as it is)
Another method is reviewing the 100+ packages that we currently have on the repository: for example, there are sets of packages that share similar characteristics, and I'd expect the packages in a certain set to look alike.

For the software part, you can trust spksrc to do the right thing...up to a certain point.
Other than that, you should use whatever the software provides to make testing easier, whether it's running automated tests, doing some manual checks based off of upstream docs, or actually _using_ the software (e.g. here you can pretend you're a Flexget user, and configure it to follow a series and whatever else it's supposed to do).

This is one of the reasons why I am ridiculously strict when I review PR's: I can only see what I am presented when someone opens a PR, but I can't tell what _you_ have done (other than filling out the checklist in the PR template) :)

**Comment by Dr-Bean** [comment]
Having said all that, if you (or anyone for that matter) have suggestions, I'm happy to hear it. Use one of the existing open issues on documentation, or just open a new one :)

**Comment by ymartin59** [comment]
Thanks for this overview. Here is a suggestion: create an additional packages dedicated to deploy and run tests on target system. I find this option better than my first idea to add a "with tests" switch to existing package  so that dependencies and dsm-control are "extended" to run and log tests (now find it quite ugly and error-prone).

In case of Flexget, "flexget-tests" package should provide additional wheels (mock vcrpy pytest pytest-capturelog boto3) to flexget virtual env, embed test scripts, and its dsm-control "start" runs "py.test" which results are collected into logs.

That way, it is possible to collect test scripts from varia sources: upstream project repository or additional dedicated scripts target specific issue found on synology for instance like checking fixed bugs to avoid regression.

**Comment by ymartin59** [comment]
@Dr-Bean Here is flexget package update for review. I am giving a try to my "flexget-tests" concept package.

File: spk/flexget/Makefile [hunk 2]
```diff
@@ -1,16 +1,18 @@
 SPK_NAME = flexget
-SPK_VERS = $(shell date +%Y%m%d)
-SPK_REV = 5
+SPK_VERS = 2.9.14
+SPK_REV = 6
 SPK_ICON = src/flexget.png
 
+# Python wheels
+# https://github.com/SynoCommunity/spksrc/wiki/Using-wheels-to-distribute-Python-packages
```
Comment (Dr-Bean) [inline spk/flexget/Makefile]: Are these two lines really necessary? It's not likely you'll forget, will you? :)

**Comment by Dr-Bean** [comment]
Looks like you've been able to generate the list of wheels yourself :) Other than the one comment, good to go.

Did you actually do the tests you mentioned for this package, other than checking if Flexget runs?
As for the discussion, let's move it to a separate issue. The -testing packages have a different purpose (it's the upstream testing versions that people want to use. From my pov, it doesn't look like people would care for doing 'internal' testing, and I think you'd struggle to get meaningful numbers out of that approach. Maybe I'm wrong though)

File: spk/flexget/Makefile (hunk 2 above)
Comment (ymartin59) [inline spk/flexget/Makefile]: No, documentation is never necessary. As I would have been happy to find such lines there at first contact to help me spare time, I wrote them thinking at future next maintainer...

Comment (ymartin59) [inline spk/flexget/Makefile]: In fact my fault, I would have read carefully https://github.com/SynoCommunity/spksrc/wiki/Developers-HOW-TO before getting into that task

Comment (Dr-Bean) [inline spk/flexget/Makefile]: Kinda, yeah :P If you feel more comfortable with it, I don't mind. Let's make it a one-liner though, so drop L6/merge into L7. Good?

Comment (Dr-Bean) [inline spk/flexget/Makefile]: oh...did you already update the PR?

**Comment by Dr-Bean** [comment]
Merged with 6b9ce3b32c6dd188e93d4fa41a7d6a378c3269bf. Package should become available in Package Center shortly. Thanks @ymartin59, much appreciated :)

**Pull request merged**